File: Programmers_python/다리를_지나는_트럭.py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    bridge = deque([0]) * bridge_length
    bridge_sum = 0
    time = 0
    
    while bridge:
        time += 1
        
        bridge_sum -= bridge.popleft()
        
        if truck_weights:
            if truck_weights[0] + bridge_sum <= weight:
                bridge_sum += truck_weights[0]
                bridge.append(truck_weights.pop(0))
            else:
                bridge.append(0)
        
    return time

File: 다리를_지나는_트럭.js
function solution(bridge_length, weight, truck_weights) {
    let answer = 0;
    let bridge = Array.from({length: bridge_length}, () => 0);
    let bridge_sum = 0;
    
    while(bridge.length > 0) {
        answer += 1;
        bridge_sum -= bridge.shift();
        
        if(truck_weights.length > 0) {
            if(bridge_sum + truck_weights[0] <= weight) {
                bridge_sum += truck_weights[0]; 
                bridge.push(truck_weights.shift());
            } else {
                bridge.push(0);
            }
        }
    }
    
    return answer;
}
